outlook/src/taskpane/components/Login/Login.tsx

```diff
@@ -45,76 +68,157 @@ class Login extends React.Component<{}, LoginState> {
         return url.origin;
     }
 
-    login = () => {
-        let sanitizedURL = '';
+    login = async () => {
+        let sanitizedURL = null;
         try {
             sanitizedURL = this.sanitizeDbUrl(this.state.baseURL);
-        } catch (err) {
-            this.setState({urlError: 'The database URL is invalid. It should follow the scheme \'https://mycompany.odoo.com\''})
+        } catch {
+            this.setState({urlState: UrlState.NotYetChecked, urlError: UrlError.InvalidScheme});
             return;
         }
 
-        //const context = React.useContext(AppContext);
+        this.setState({
+            baseURL: sanitizedURL,
+            urlState: UrlState.Checking,
+        });
+
         api.baseURL = sanitizedURL;
-        localStorage.setItem('baseURL', sanitizedURL)
+        localStorage.setItem('baseURL', sanitizedURL);
+
+        try {
+            await this._checkDatabaseUrl();
+        } catch {
+            this.setState({urlState: UrlState.NotYetChecked, urlError: UrlError.DatabaseNotReachable});
+            return;
+        }
+
+        this.setState({urlState: UrlState.NotYetChecked, urlError: UrlError.None});
+
+        let authCode = null;
+        try {
+            authCode = await this._openOdooLoginDialog();
```

Comment:
Same here, could have the underlying implementation resolve(false) on failure and check:
```js
const authCode = await this._openOdooLoginDialog();
if (!authCode) {
    // blah
}
```

And same for the exchange here below.

Reply:
Good idea :) Good is indeed a lot cleaner